scm-ui/ui-webapp/src/admin/plugins/containers/PluginsOverview.tsx

```diff
@@ -35,89 +34,50 @@ import {
 
-type Props = WithTranslation & {
-  loading: boolean;
-  error: Error;
-  collection: PluginCollection;
-  baseUrl: string;
+type Props = {
   installed: boolean;
-  availablePluginsLink: string;
-  installedPluginsLink: string;
-  pendingPluginsLink: string;
-  pendingPlugins: PendingPlugins;
-
-  // dispatched functions
-  fetchPluginsByLink: (link: string) => void;
-  fetchPendingPlugins: (link: string) => void;
 };
 
-type State = {
-  showPendingModal: boolean;
-  showExecutePendingModal: boolean;
-  showUpdateAllModal: boolean;
-  showCancelModal: boolean;
-};
-
-class PluginsOverview extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      showPendingModal: false,
-      showExecutePendingModal: false,
-      showUpdateAllModal: false,
-      showCancelModal: false
-    };
-  }
-
-  componentDidMount() {
-    this.fetchPlugins();
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    const { installed } = this.props;
-    if (prevProps.installed !== installed) {
-      this.fetchPlugins();
-    }
-  }
-
-  fetchPlugins = () => {
-    const {
-      installed,
-      fetchPluginsByLink,
-      availablePluginsLink,
-      installedPluginsLink,
-      pendingPluginsLink,
-      fetchPendingPlugins
-    } = this.props;
-    fetchPluginsByLink(installed ? installedPluginsLink : availablePluginsLink);
-    if (pendingPluginsLink) {
-      fetchPendingPlugins(pendingPluginsLink);
-    }
-  };
-
-  renderHeader = (actions: React.ReactNode) => {
-    const { installed, t } = this.props;
+const PluginsOverview: FC<Props> = ({ installed }) => {
+  const [t] = useTranslation("admin");
+  const queryClient = useQueryClient();
+  const {
+    data: availablePlugins,
+    isLoading: isLoadingAvailablePlugins,
+    error: availablePluginsError
+  } = useAvailablePlugins(!installed);
+  const {
+    data: installedPlugins,
+    isLoading: isLoadingInstalledPlugins,
+    error: installedPluginsError
+  } = useInstalledPlugins(installed);
+  const { data: pendingPlugins, isLoading: isLoadingPendingPlugins, error: pendingPluginsError } = usePendingPlugins();
+  const [showPendingModal, setShowPendingModal] = useState<boolean>();
+  const [showExecutePendingModal, setShowExecutePendingModal] = useState<boolean>();
+  const [showUpdateAllModal, setShowUpdateAllModal] = useState<boolean>();
+  const [showCancelModal, setShowCancelModal] = useState<boolean>();
+  const collection = installed ? installedPlugins : availablePlugins;
+  const error = (installed ? installedPluginsError : availablePluginsError) || pendingPluginsError;
+  const loading = (installed ? isLoadingInstalledPlugins : isLoadingAvailablePlugins) || isLoadingPendingPlugins;
+
+  const refresh = () =>
+    Promise.all([
+      queryClient.invalidateQueries("availablePlugins"),
+      queryClient.invalidateQueries("installedPlugins"),
+      queryClient.invalidateQueries("pendingPlugins")
+    ]);
```

Comment:
Why do we need this? If we really need this, we should restructure the query keys in something like `["plugins", "installed"]`.